src/runtime/gravity_core.c

```diff
@@ -1450,6 +1450,171 @@ static bool operator_string_cmp (gravity_vm *vm, gravity_value_t *args, uint16_t
+	for (int i = 0; i < main_str->len; i++) {
+		if (main_str->s[i] == str_to_count->s[j]) {
+			// If the characters match and we are on the last character of the search
+			// string, then we have found a match
+			if (j == str_to_count->len - 1) {
+				count++;
+				j = 0;
+				continue;
+			}
+		}
+		// Reset if it isn't a match
+		else {
+			j = 0;
+			continue;
+		}
+		// Move forward in the search string if we found a match but we aren't
+		// finished checking all the characters of the search string yet
+		j++;
+	}
+
+	RETURN_VALUE(VALUE_FROM_INT(count), rindex);
+}
+
+
+static bool string_repeat (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	#pragma unused(vm, nargs)
+	gravity_string_t *main_str = VALUE_AS_STRING(GET_VALUE(0));
+
+	if (nargs != 2) {
+		RETURN_ERROR("String.string_repeat() expects 1 argument");
+	}
+	if (!VALUE_ISA_INT(GET_VALUE(1))) {
+		RETURN_ERROR("String.string_repeat() expects an integer as an argument");
+	}
+	gravity_int_t times_to_repeat = VALUE_AS_INT(GET_VALUE(1));
+
+	if (times_to_repeat < 1) {
+		RETURN_ERROR("String.string_repeat() expects an integer >=1");
+	}
+
+	// Figure out the size of the array we need to make to hold the new string
+	int new_size = main_str->len*times_to_repeat;
+	char new_str[new_size];
+	strcpy(new_str, main_str->s);
+	for (int i = 0; i < times_to_repeat-1; i++) {
+		strcat(new_str, main_str->s);
+	}
+
+	RETURN_VALUE(VALUE_FROM_STRING(vm, new_str, strlen(new_str)), rindex);
+}
+
+static bool string_upper (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	#pragma unused(vm, nargs)
+	gravity_string_t *main_str = VALUE_AS_STRING(GET_VALUE(0));
+
+	if (nargs != 1) {
+		RETURN_ERROR("String.upper() expects no argument");
+	}
+
+	for (int i = 0; i < main_str->len; i++) {
+		main_str->s[i] = toupper(main_str->s[i]);
+	}
+
+	RETURN_VALUE(VALUE_FROM_STRING(vm, main_str->s, main_str->len), rindex);
+}
+
+static bool string_lower (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	#pragma unused(vm, nargs)
+	gravity_string_t *main_str = VALUE_AS_STRING(GET_VALUE(0));
+
+	if (nargs != 1) {
+		RETURN_ERROR("String.lower() expects no argument");
+	}
+
+	for (int i = 0; i < main_str->len; i++) {
+		main_str->s[i] = tolower(main_str->s[i]);
+	}
+
+	RETURN_VALUE(VALUE_FROM_STRING(vm, main_str->s, main_str->len), rindex);
+}
+
+static bool string_loadat (gravity_vm *vm, gravity_value_t *args, uint16_t nargs, uint32_t rindex) {
+	#pragma unused(vm, nargs)
+	gravity_string_t *string = VALUE_AS_STRING(GET_VALUE(0));
+	gravity_value_t value = GET_VALUE(1);
+	if (!VALUE_ISA_INT(value)) RETURN_ERROR("An integer index is required to access a string item.");
+	
+	register int32_t index = (int32_t)VALUE_AS_INT(value);
+	
+	if (index < 0) index = string->len + index;
+	if ((index < 0) || ((uint32_t)index >= string->len)) RETURN_ERROR("Out of bounds error: index %d beyond bounds 0...%d", index, string->len-1);
+
+	char c[2] = "\0";
```

Comment:
I am not sure specifically what you are referring to, but everything here looks like it should be there.

That index check is to allow negative indices to be used to start from the end of the string (it is code that is used elsewhere as well), and line 1589 is setting up a single character string... since replacing c[0] with a character (for example "x") makes `c == "x", due to the \0 null terminator.